src/TRoom.cpp

```diff
@@ -799,6 +800,12 @@ void TRoom::restore(QDataStream& ifs, int roomID, int version)
             ifs >> customLinesArrow;
             ifs >> customLinesColor;
             ifs >> customLinesStyle;
+            if (version >= 21) {
+                qreal x, y;
+                ifs >> x;
+                ifs >> y;
+                nameOffset = QPoint(x, y);
```

Comment:
Don't you need `nameOffset = QPointF(x, y);` here - `x` and `y` are floating point numbers and `QPoint` uses (faster) integers only.